Source/Provers/SMTLib/ProverInterface.cs

```diff
@@ -2654,6 +2656,17 @@ public override void SetRlimit(int limit)
       }
     }
 
+    public override void SetRandomSeed(int randomSeed)
+    {
+      if (options.Solver == SolverKind.Z3)
+      {
+        var name = Z3.RandomSeedOption;
+        var value = randomSeed.ToString();
+        options.SmtOptions.RemoveAll(ov => ov.Option == name);
+        options.AddSmtOption(name, value);
```

Comment:
`options` (which is of type `SMTLibProverOptions`) has methods `AddSmtOption` which just adds to the list of options, and `AddWeakSmtOption` which only adds an option if no entry for the same name exists yet. Maybe we should add a third method for adding/overriding options (encapsulating the call to `RemoveAll` here and in `SetTimeout` and `SetRlimit` above).

Reply:
These method APIs appear strange to me.  There should be just one method SetSmtOption, which should either add or override as the case might be.  This is easy and fast to do if I change the representation from a list to a dictionary.

Reply:
Timeout, Rlimit, and RandomSeed are not put into the SmtOptions list any more.  They are set directly inside BeginCheck now.